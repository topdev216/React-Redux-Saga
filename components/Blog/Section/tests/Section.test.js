import React from 'react';
import { shallow } from 'enzyme';
import { Map, List } from 'immutable';

import { BLOG_LATEST_KEY, BLOG_LATEST_TITLE } from 'containers/constants';
import BlogSection from 'components/Blog/Section';

describe('<BlogSection />', () => {
  it('should match snapshoot', () => {
    const renderedComponent = shallow(
      <BlogSection
        slug="category"
        title="Category"
        total={4}
        displayedPosts={new List([new Map(), new Map(), new Map()])}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have the title copy', () => {
    const renderedComponent = shallow(
      <BlogSection
        slug="category"
        title="Category"
        total={4}
        displayedPosts={new List([new Map(), new Map(), new Map()])}
      />
    );
    expect(renderedComponent.find('h2').text()).toEqual('Category');
  });

  it('should pass the correct props to PostList for latest category', () => {
    const renderedComponent = shallow(
      <BlogSection
        title={BLOG_LATEST_TITLE}
        slug={BLOG_LATEST_KEY}
        total={4}
        displayedPosts={new List([new Map(), new Map(), new Map()])}
      />
    );
    expect(renderedComponent.find('PostList')).toHaveLength(1);
    expect(renderedComponent.find('PostList').prop('chunkStyle')).toBeFalsy();
    expect(
      renderedComponent.find('PostList').prop('jumbotronStyle')
    ).toBeTruthy();
  });

  it('should pass the correct props to PostList for other category', () => {
    const renderedComponent = shallow(
      <BlogSection
        slug="category"
        title="Category"
        total={4}
        displayedPosts={new List([new Map(), new Map(), new Map()])}
      />
    );
    expect(renderedComponent.find('PostList')).toHaveLength(1);
    expect(renderedComponent.find('PostList').prop('chunkStyle')).toBeTruthy();
    expect(
      renderedComponent.find('PostList').prop('jumbotronStyle')
    ).toBeFalsy();
  });

  it('should have a Load More button when there are still more posts', () => {
    const renderedComponent = shallow(
      <BlogSection
        slug="category"
        title="Category"
        total={4}
        displayedPosts={new List([new Map(), new Map(), new Map()])}
      />
    );
    expect(renderedComponent.find('Button')).toHaveLength(1);
    expect(renderedComponent.find('Button').prop('element')).toEqual('button');
  });

  it('should not render the Load More button when all posts are displayed', () => {
    const renderedComponent = shallow(
      <BlogSection
        slug="category"
        title="Category"
        total={4}
        displayedPosts={new List([new Map(), new Map(), new Map(), new Map()])}
      />
    );
    expect(renderedComponent.find('Button')).toHaveLength(0);
  });
});
