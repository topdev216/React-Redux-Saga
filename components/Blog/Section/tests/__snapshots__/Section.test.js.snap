// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogSection /> should match snapshoot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="blogSection row column"
>
    <header>
        <h2
            className="blogSection__title"
        >
            Category
        </h2>
    </header>
    <PostList
        chunkStyle={true}
        data={
            Immutable.List [
                Immutable.Map {
                },
                Immutable.Map {
                },
                Immutable.Map {
                },
            ]
        }
        isLoading={false}
        jumbotronStyle={false}
        noPostsMessage="No posts here"
        showCategory={true}
    />
    <Button
        className="expanded secondary large mb-xl"
        element="button"
        onClick={undefined}
    >
        Read more 
        Category
    </Button>
</section>,
  "nodes": Array [
    <section
      className="blogSection row column"
>
      <header>
            <h2
                  className="blogSection__title"
            >
                  Category
            </h2>
      </header>
      <PostList
            chunkStyle={true}
            data={
                  Immutable.List [
                        Immutable.Map {
                        },
                        Immutable.Map {
                        },
                        Immutable.Map {
                        },
                  ]
            }
            isLoading={false}
            jumbotronStyle={false}
            noPostsMessage="No posts here"
            showCategory={true}
      />
      <Button
            className="expanded secondary large mb-xl"
            element="button"
            onClick={undefined}
      >
            Read more 
            Category
      </Button>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogSection
        displayedPosts={
                Immutable.List [
                        Immutable.Map {
                        },
                        Immutable.Map {
                        },
                        Immutable.Map {
                        },
                ]
        }
        slug="category"
        title="Category"
        total={4}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "displayedPosts": Immutable.List [
            Immutable.Map {
            },
            Immutable.Map {
            },
            Immutable.Map {
            },
],
          "slug": "category",
          "title": "Category",
          "total": 4,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="blogSection row column"
>
          <header>
                    <h2
                              className="blogSection__title"
                    >
                              Category
                    </h2>
          </header>
          <PostList
                    chunkStyle={true}
                    data={
                              Immutable.List [
                                        Immutable.Map {
                                        },
                                        Immutable.Map {
                                        },
                                        Immutable.Map {
                                        },
                              ]
                    }
                    isLoading={false}
                    jumbotronStyle={false}
                    noPostsMessage="No posts here"
                    showCategory={true}
          />
          <Button
                    className="expanded secondary large mb-xl"
                    element="button"
                    onClick={undefined}
          >
                    Read more 
                    Category
          </Button>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="blogSection row column"
>
          <header>
                    <h2
                              className="blogSection__title"
                    >
                              Category
                    </h2>
          </header>
          <PostList
                    chunkStyle={true}
                    data={
                              Immutable.List [
                                        Immutable.Map {
                                        },
                                        Immutable.Map {
                                        },
                                        Immutable.Map {
                                        },
                              ]
                    }
                    isLoading={false}
                    jumbotronStyle={false}
                    noPostsMessage="No posts here"
                    showCategory={true}
          />
          <Button
                    className="expanded secondary large mb-xl"
                    element="button"
                    onClick={undefined}
          >
                    Read more 
                    Category
          </Button>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogSection
    displayedPosts={
        Immutable.List [
            Immutable.Map {
            },
            Immutable.Map {
            },
            Immutable.Map {
            },
        ]
    }
    slug="category"
    title="Category"
    total={4}
/>,
}
`;
